### PR TITLE
initial setup for localization

### PR DESCRIPTION
**Pull request by FloVanGH**
# Context:

* localization sub crate
* localization trait
* default localization implementation based on `ron`
* localize string properties in OrbTk
* add example to showcase 

#113 

@mentions of the person or team responsible for reviewing proposed changes.

## Contribution checklist:
- [x] Add [documentation](https://doc.rust-lang.org/1.7.0/book/documentation.html) to all public structs, traits and functions.
- [x] Add unit tests if possible
- [x] Describe the major change(s) in the CHANGELOG.MD
- [x] Run `cargo fmt` to make the formatting consistent across the codebase
- [x] Run `cargo clippy` to check with the linter

### CODE REVIEW

File: crates/localization/src/localization.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+///
+pub trait Localization {
+    /// Gets the current language by language key e.g. `en_US` or `de_DE`.
+    fn language(&self) -> &String;
+
+    /// Sets the current language by key e.g. `en_US` or `de_DE`.
+    fn set_language(&mut self, key: &str);
```
Comment (kivimango) [inline crates/localization/src/localization.rs:7]: I think the Localization struct should be immutable.

File: crates/localization/src/localization.rs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    /// Gets the translated text for the given key. If there is no given translation for the given key, the key will be returned as string.
+    fn text(&self, key: &str) -> String;
```
Comment (kivimango) [inline crates/localization/src/localization.rs:10]: maybe the get_text() name is more convinient ?

**Comment by kivimango** [comment]
Great kickstart and motivation to contribute!
I would do it, but I am overthinking the design for weeks.

**Comment by FloVanGH** [comment]
We discussed today with a customer the possibility of localization and after that I had an idea how to handle it in OrbTk.

The localization will be optional. An expulsion cab create your custom localization based on the `Locializaton` trait. 

You will be able to set a flag on text widgets of it is localizable. Default on `TextBlock` would be `true`, on `TextBox` `false` and on `PasswordBox` it can not be set it will be `false`. 

And I think of defining an UI with OrbTk in a chosen language e.g. English. And the set text will be the key in the language file. You can choose if you provide a language file for your default language or not. If not the given `text` string will be displayed without translation. This will also not break the current API. And you can also start with one language without thinking of localization and add it later.

**Comment by FloVanGH** [comment]
Before that idea I was also thinking about this topic for months 🙈

**Comment by adals** [comment]
Thank you @FloVanGH  for your hard work. Appreciated 

what about RTL languages Like Arabic?

thanks

**Comment by FloVanGH** [comment]
@adals thank you. Can you do me a favor and create an issue for RTL language feature request?

@sandmor you are working on text input. Do you think you can consider right to left text drawing and input on your work? I think direction would be the first step. Draw the right glyphs with a font the second.

**Comment by adals** [comment]
@FloVanGH kindly see [330](https://github.com/redox-os/orbtk/issues/330)

File: crates/api/src/layout/fixed_size.rs [hunk 3]
```diff
@@ -194,3 +194,19 @@ impl Layout for FixedSizeLayout {
...
+            }
+        }
+    }
+
+    if let Some(text) = widget.try_get::<String>("text") {
+        return Some(text.clone());
+    }
+
+    None
+}
```
Comment (kivimango) [inline crates/api/src/layout/fixed_size.rs:212]: I used to work with PHP and Java HTML template engines. Those used the string interpolation technique, like
`{#text}` String would be replaced by the value in a dictionary of the corresponding locale.
The locale or the `{#text}` property could be the key to the value in the dictionary.
This way we could get rid of the `localizable` property.

File: crates/localization/src/ron_localization/mod.rs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        "#;
+
+        let localization = RonLocalization::create()
+            .language("de_DE")
+            .dictionary("de_DE", de_de)
+            .build();
+
+        assert_eq!(localization.text("hello".to_string()), "Hallo".to_string());
+        assert_eq!(localization.text("world".to_string()), "Welt".to_string());
+        assert_eq!(localization.text("test".to_string()), "test".to_string());
```
Comment (kivimango) [inline crates/localization/src/ron_localization/mod.rs:106]: Could you please add the "sad" cases to tests? Like when the given is not found in the dictionary.

**Review by kivimango — APPROVED** [review]
I think this approach is OK, it fits into the "orbtk" way.

**Comment by FloVanGH** [comment]
I explicit do it not this way, that you can write you ui complete without localization in mind (in one language). If you want to add other languages you can register a localization service and use the normal text Strings as key.

I see this a start point and further ideas and contributions in this are welcome.

File: crates/localization/src/ron_localization/mod.rs (hunk 4 above)
Comment (FloVanGH) [reply crates/localization/src/ron_localization/mod.rs:106]: test is the sad path. If there is no key the key will returned as result.

File: crates/api/src/layout/fixed_size.rs (hunk 3 above)
Comment (FloVanGH) [reply crates/api/src/layout/fixed_size.rs:212]: I explicit do it not this way, that you can write you ui complete without localization in mind (in one language). If you want to add other languages you can register a localization service and use the normal text Strings as key.

I see this a start point and further ideas and contributions in this are welcome.

**Comment by FloVanGH** [comment]
@adals thank you

File: crates/localization/src/ron_localization/mod.rs (hunk 4 above)
Comment (kivimango) [reply crates/localization/src/ron_localization/mod.rs:106]: oh, somehow slipped over this...

File: crates/api/src/layout/fixed_size.rs (hunk 3 above)
Comment (kivimango) [reply crates/api/src/layout/fixed_size.rs:212]: i see...

**Review by kivimango — APPROVED** [review]

**Pull request merged**